src/tests/Interop/COM/ComWrappers/API/Program.cs

```diff
@@ -336,6 +336,62 @@ unsafe static void CallSetValue(TestComWrappers wrappers, Test testInstance, int
             }
         }
 
+        [MethodImpl(MethodImplOptions.NoInlining)]
+        [Fact]
+        public void ValidateResurrection()
+        {
+            Console.WriteLine($"Running {nameof(ValidateResurrection)}...");
+
+            var wrappers = new TestComWrappers();
+
+            try
+            {
+                CreateResurrectingTestInstance(wrappers);
+
+                ForceGC();
+
+                CallSetValue(wrappers);
+            }
+            finally
+            {
+                Test.Resurrected = null;
+            }
+
+            [MethodImpl(MethodImplOptions.NoInlining)]
+            static void CreateResurrectingTestInstance(ComWrappers wrapper)
+            {
+                Test testInstance = new Test()
+                {
+                    EnableResurrection = true,
+                };
+                IntPtr nativeInstance = wrapper.GetOrCreateComInterfaceForObject(testInstance, CreateComInterfaceFlags.None);
+                Assert.Equal(0, Marshal.Release(nativeInstance));
+            }
+
+            unsafe static void CallSetValue(ComWrappers wrappers)
+            {
+                Assert.NotEqual(null, Test.Resurrected);
+                IntPtr nativeInstance = wrappers.GetOrCreateComInterfaceForObject(Test.Resurrected, CreateComInterfaceFlags.None);
+                Assert.NotEqual(IntPtr.Zero, nativeInstance);
+
+                var iid = typeof(ITest).GUID;
+                IntPtr itestPtr;
+                Assert.Equal(0, Marshal.QueryInterface(nativeInstance, ref iid, out itestPtr));
+
+                var inst = Marshal.PtrToStructure<VtblPtr>(itestPtr);
```

Comment:
We shouldn't be using `Marshal.PtrToStructure()` anywhere in here. Please defer to pointers and remove the other two uses as well.

Reply:
I switch to using pointers.